torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -354,11 +354,41 @@ void TensorPipeAgent::prepareNames() {
   }
 }
 
+void TensorPipeAgent::checkAndSetStaticGroup(
+    const c10::intrusive_ptr<::c10d::Store>& store) {
+  std::string isStaticGroupKey("rpcIsStaticGroup");
+
+  std::string isStaticGroupStr = isStaticGroup_ ? "true" : "false";
+  std::vector<uint8_t> isStaticGroupVec(
+      (uint8_t*)isStaticGroupStr.c_str(),
+      (uint8_t*)isStaticGroupStr.c_str() + isStaticGroupStr.length());
+  std::vector<uint8_t> returnedVec;
+  bool isStaticGroupKeyExists =
+      store->check(std::vector<std::string>{isStaticGroupKey});
+  if (isStaticGroupKeyExists) {
+    returnedVec =
+        store->compareSet(isStaticGroupKey, isStaticGroupVec, isStaticGroupVec);
+  } else {
+    returnedVec = store->compareSet(
+        isStaticGroupKey, std::vector<uint8_t>(), isStaticGroupVec);
+  }
+  std::string returnedVal = std::string(returnedVec.begin(), returnedVec.end());
+  // In both cases, the returned value should be the value of isStaticGroupStr,
+  // otherwise there is a discrepency with initialization among one of the
+  // members
+  TORCH_CHECK(
+      returnedVal == isStaticGroupStr,
+      fmt::format(
+          "RPC group behavior is different than expected. isStaticGroup_ is initialized with {} while in store it is {}",
```

Comment:
Is it possible to add a test to cover this? I assume not, because the static code path would actually create a process group before creating TP agent instances?